resource-managers/mesos/src/main/java/org/apache/spark/network/shuffle/mesos/MesosExternalBlockStoreClient.java

```diff
@@ -89,7 +89,7 @@ private RegisterDriverCallback(TransportClient client, long heartbeatIntervalMs)
 
     @Override
     public void onSuccess(ByteBuffer response) {
-      heartbeaterThread.scheduleAtFixedRate(
+      heartbeaterThread.scheduleWithFixedDelay(
           new Heartbeater(client), 0, heartbeatIntervalMs, TimeUnit.MILLISECONDS);
       logger.info("Successfully registered app " + appId + " with external shuffle service.");
```

Comment:
I am not sure who is working on mesos to understand impact of this behavior change.
+CC @vanzin, @tgravescs ?

I would prefer preserving behavior if someone cant ack it is fine to modify this.